packages/marketplace/docs/exchange/Exchange.md

```diff
@@ -0,0 +1,170 @@
+# Exchange
+
+## Introduction
+
+The `Exchange` contract is the entrypoint and main contract to the marketplace protocol.
+It safely offers a decentralized way to exchange tokens of any nature (ERC20, ERC1155, ERC721) using signed orders.
+
+## Concepts
+
+### Order
+
+An exchange consists of a trade between two parties. Each side of the trade is called an order. That order contains all the information required to define what a party is asking for. Let's consider this use case:
+
+Order A
+```
+Alice wants to sell to anybody 1 LAND (ERC721) with token id 1000 against 100 SAND (ERC20).
+```
+
+The order represents this intent and includes the address of the seller, the address of the NFT, the token id, the address of the ERC20 and the amount the seller is asking for.
+
+### Maker & Taker
+
+The maker and taker represent the 2 parties of the exchange. Each order can define what should contain each party where the maker represent your side and the taker the other side.
+
+For instance, as a seller, I can decide to sell a NFT to anybody. But, I could also specify that I only want to sell it to a particular user. In order to do that, you have to input the taker address.
+Same reasoning, if I want to buy a NFT but I don't mind which seller is selling me the NFT, all I have to do is creating an order without a taker address. Otherwise, if I precisely want to buy it from a seller, I can specify his address in the order.
+
+### Matching
+
+To execute a trade, you need two parties (and so 2 orders) that match. So, the contract takes in 2 orders and check that both orders are actually a match and satisfy each party. For instance, let's take again our seller use case:
+
+`Order A`
+```
+Alice wants to sell to anybody 1 LAND (ERC721) with token id 1000 against 100 MATIC (ERC20).
+```
+
+An order satisfying Order A could simply be:
+
+`Order B`
+```
+Bob wants to buy from anybody 1 LAND (ERC721) with token id 1000 against 100 MATIC (ERC20).
+```
+
+### Validation
+
+In order to validate the maker of each order, the contract supports two ways to validate an order:
+- the sender of the transaction has to be the maker of the order
+- the sender has to provide the signature of the order signed by the order's maker
+
+It brings a powerful asynchronous system but also users should be careful when signing an order because anybody can send a transaction to match 2 orders if that user gets the signatures.
+For instance:
+- Alice signs `Order A`
+- Bob signs `Order B`
+- Carol executes the trades with the 2 orders `Order A` and `Order B` and the 2 signatures
```

Comment:
Just need to clarify here whether Carol is doing a bad thing. And maybe we can just add a sentence on how we've tried to make it safe?

Reply:
suggestion
```
- Alice signs `Order A` and lends the signature to Carol
- Bob signs `Order B` and lends the signature to Carol
- Carol executes the trades with the 2 orders `Order A` and `Order B` and the 2 signatures
- Alice & Bob must trust Carol
```